src/sentry/sentry_apps/services/app/impl.py

```diff
@@ -171,6 +171,42 @@ def get_component_contexts(
                 output.append(context_item)
         return output
 
+    def get_installation_component_contexts(
+        self,
+        *,
+        filter: SentryAppInstallationFilterArgs,
+        component_type: str,
+        include_contexts_with_component: bool = True,
+    ) -> list[RpcSentryAppComponentContext]:
+        install_query = self._FQ.query_many(filter=filter)
+        app_ids = install_query.select_related("sentry_app", "sentry_app__application").values_list(
+            "sentry_app_id", flat=True
+        )
+        component_query = SentryAppComponent.objects.filter(
+            type=component_type, sentry_app_id__in=list(app_ids)
+        )
+        component_map: dict[int, SentryAppComponent] = {}
+        output = []
+
+        for component in component_query:
+            component_map[component.sentry_app.id] = component
+
+        for install in install_query:
+            install_component = component_map.get(install.sentry_app_id)
+            if include_contexts_with_component or install_component:
```

Comment:
The `include_contexts_with_component` variable name is a little confusing. So this boolean flag only skips adding contexts that don't have components defined when it's set to `False`?

Reply:
shoot this made sense to me last night but you're right i need to fix this 😅